### ISSUE TITLE
Top sloping surfaces aren't supported well enough

### ISSUE THREAD

**Issue by hyperair** [alexrj/Slic3r#2338]
This seems to be similar to #864, but is not quite, because I'm not getting gaps.

When printing with very thin layer heights (0.1mm), I like to increase the number of top shells to 10 or 20 so that I get a 1-2mm thick top layer. This smooths over any problems caused by improper bridging over sparse infill, so you get a very nice and solid upper layer when you're done.

With relatively flat sloped surfaces however (10°), what happens is that Slic3r doesn't properly support the "overhung" perimeters. By "overhung", I mean the perimeter layers that are printed partially on the lower layer's perimeter, and partially dangling on top of infill. At this point, I think Slic3r is supposed to dynamically increase the number of perimeters to avoid gaps, but it doesn't seem to be enough to guarantee a proper 1mm thick top layer, so the typical insufficient-infill artifacts start showing up, e.g. the top surface bulges up slightly where it isn't touching an infill line.

In cases like this, Slic3r should probably handle it better by increasing the number of perimeters such that there are at least ($top_solid_layers - 1) layers supporting the top partially-overhung perimeters. This should also have the effect of closing any gaps as found in #864.

**Comment by more10** [comment]
I have exactly the same problem, at bottam and at top.

**Comment by lordofhyphens** [comment]
@hyperair can you post a config you are getting this problem on? Zipped is fine.

**Comment by lordofhyphens** [comment]
According to @VanessaE disabling "Only infill where needed" clears the symptom.

**Comment by hyperair** [comment]
> According to @VanessaE disabling "Only infill where needed" clears the symptom.

I never have that option enabled.

**Comment by hyperair** [comment]
Here's some sample G-Code. Seems like it's layer-height dependent. At 0.1mm, the issue happens, but with 0.2mm, it puts enough extra layers underneath.

Here's 0.1mm: https://gist.github.com/2164b315b3d5a58e15669c56ecc0d23c
![screenshot16](https://cloud.githubusercontent.com/assets/88355/15984600/2899bb48-3004-11e6-9469-d510298a17b4.png)

Here's 0.2mm: https://gist.github.com/cf273ca18b254c2a0237c2c1d5359080
![screenshot17](https://cloud.githubusercontent.com/assets/88355/15984612/8a4c8410-3004-11e6-9712-8d4bba77a857.png)

**Comment by hyperair** [comment]
Oh yeah I forgot to mention: I'm using 209b919. Config's in the gcode. Extract it with:
```sh
sed -n '/filament used/,$p' foo.gcode | tail -n+3 | sed -e 's,^; ,,' | less
```

**Comment by VanessaE** [comment]
Indeed, I was wrong about the "only where needed" option, as I'm experiencing this issue now as well.  It was just well hidden with the models I've been printing up till now.

**Comment by lordofhyphens** [comment]
Does #3610 solve the issue sufficiently?

**Comment by bubnikv** [comment]
no, but the "ensure vertical wall thickness" feature of the prusa3d fork does.

**Comment by alexrj** [comment]
This was probably fixed by 11ed3d8cf3fd1713b465604535a3a0403cd6d0ad (Dec 19).

**Comment by lordofhyphens** [comment]
Looks correct now to me @ 0.1 
![screenshot - 03102017 - 11 57 12 pm](https://cloud.githubusercontent.com/assets/31754/23821048/57a5d628-05ed-11e7-880c-096e2a4ad35a.png)
![screenshot from 2017-03-10 23 59 00](https://cloud.githubusercontent.com/assets/31754/23821064/95ec6654-05ed-11e7-8b5d-77142b9dcf4e.png)
